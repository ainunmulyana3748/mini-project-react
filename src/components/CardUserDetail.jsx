import React, { useState } from "react";
import { ArrowLeft, Star } from "lucide-react";
import { useNavigate } from "react-router-dom";
import SuccessModal from "./SuccessModal";

const CardUserDetail = ({ user, button }) => {
  const navigate = useNavigate();
  const [showModal, setShowModal] = useState(false);
  const rating = 4.5;
  const languages = ["English", "Bahasa Indonesia", "French"];

  return (
    <>
      <div className="w-full max-w-xl mx-auto bg-white rounded-3xl shadow-lg p-6 sm:p-8 mt-6 transition-all duration-300 hover:shadow-blue-100">
        {/* Back Button */}
        <ArrowLeft
          size={28}
          className="cursor-pointer mb-4 text-gray-600 hover:text-blue-600 transition"
          onClick={() => navigate(-1)}
        />

        <div className="flex flex-col sm:flex-row items-center sm:items-start gap-6">
          {/* Avatar */}
          <img
            src={user.avatar}
            alt={user.first_name}
            className="w-32 h-32 sm:w-36 sm:h-36 rounded-full border-4 border-blue-500 shadow-md object-cover"
          />

          {/* User Info */}
          <div className="flex-1 text-center sm:text-left">
            <h2 className="text-2xl sm:text-3xl font-bold text-gray-800">
              {user.first_name} {user.last_name}
            </h2>
            <p className="text-gray-500 text-sm mt-1">{user.email}</p>

            {/* Rating */}
            <div className="flex items-center justify-center sm:justify-start gap-1 mt-3">
              {Array.from({ length: 5 }, (_, i) => (
                <Star
                  key={i}
                  size={18}
                  className={
                    i < Math.floor(rating)
                      ? "text-yellow-400 fill-yellow-400"
                      : "text-yellow-300"
                  }
                />
              ))}
              <span className="ml-2 text-sm text-gray-600">{rating}/5.0</span>
            </div>

            {/* Info Detail */}
            <div className="mt-5 text-sm text-gray-700 space-y-2">
              <p>
                <span className="font-semibold">Role:</span> Tour Guide
              </p>
              <p>
                <span className="font-semibold">Status:</span> Active
              </p>
              <p>
                <span className="font-semibold">Tour Guide Since:</span> 2024
              </p>
              <p>
                <span className="font-semibold">Languages:</span>{" "}
                <span className="inline-flex flex-wrap gap-2 mt-1">
                  {languages.map((lang) => (
                    <span
                      key={lang}
                      className="bg-blue-100 text-blue-800 text-xs font-medium px-2 py-1 rounded-full"
                    >
                      {lang}
                    </span>
                  ))}
                </span>
              </p>
            </div>

            <div className="mt-6">{button}</div>
          </div>
        </div>
      </div>

      {showModal && (
        <SuccessModal
          type="booking"
          title="Booking Confirmed!"
          message="Your guide has been successfully booked."
        />
      )}
    </>
  );
};

export default CardUserDetail;
