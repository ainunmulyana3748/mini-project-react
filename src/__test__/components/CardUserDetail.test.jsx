import { render, screen, fireEvent } from "@testing-library/react";
import CardUserDetail from "../../components/CardUserDetail";
import { BrowserRouter } from "react-router-dom";

// Mock useNavigate
const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

describe("CardUserDetail Component", () => {
  const user = {
    id: 2,
    avatar: "https://example.com/avatar2.jpg",
    email: "user@mail.com",
    first_name: "Siti",
    last_name: "Aisyah",
  };

  test("render nama, email, avatar, rating, dan languages", () => {
    render(
      <BrowserRouter>
        <CardUserDetail user={user} />
      </BrowserRouter>
    );
    expect(screen.getByText("Siti Aisyah")).toBeInTheDocument();
    expect(screen.getByText("user@mail.com")).toBeInTheDocument();
    expect(screen.getByAltText("Siti")).toHaveAttribute("src", user.avatar);
    expect(screen.getByText("4.5/5.0")).toBeInTheDocument();
    expect(screen.getByText("English")).toBeInTheDocument();
    expect(screen.getByText("Bahasa Indonesia")).toBeInTheDocument();
    expect(screen.getByText("French")).toBeInTheDocument();
  });

  test("navigasi back saat tombol back diklik", () => {
    render(
      <BrowserRouter>
        <CardUserDetail user={user} />
      </BrowserRouter>
    );
    const backButton = screen.getByRole("img", { hidden: true }); // ArrowLeft
    fireEvent.click(backButton);
    expect(mockedUsedNavigate).toHaveBeenCalledWith(-1);
  });

  test("render children pada props button", () => {
    render(
      <BrowserRouter>
        <CardUserDetail user={user} button={<button>Test Button</button>} />
      </BrowserRouter>
    );
    expect(screen.getByText("Test Button")).toBeInTheDocument();
  });

  test("snapshot card user detail", () => {
    const { container } = render(
      <BrowserRouter>
        <CardUserDetail user={user} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
