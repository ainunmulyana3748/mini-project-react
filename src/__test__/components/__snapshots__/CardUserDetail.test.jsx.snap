// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardUserDetail Component snapshot card user detail 1`] = `
<div>
  <div
    class="w-full max-w-xl mx-auto bg-white rounded-3xl shadow-lg p-6 sm:p-8 mt-6 transition-all duration-300 hover:shadow-blue-100"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-arrow-left cursor-pointer mb-4 text-gray-600 hover:text-blue-600 transition"
      fill="none"
      height="28"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m12 19-7-7 7-7"
      />
      <path
        d="M19 12H5"
      />
    </svg>
    <div
      class="flex flex-col sm:flex-row items-center sm:items-start gap-6"
    >
      <img
        alt="Siti"
        class="w-32 h-32 sm:w-36 sm:h-36 rounded-full border-4 border-blue-500 shadow-md object-cover"
        src="https://example.com/avatar2.jpg"
      />
      <div
        class="flex-1 text-center sm:text-left"
      >
        <h2
          class="text-2xl sm:text-3xl font-bold text-gray-800"
        >
          Siti
           
          Aisyah
        </h2>
        <p
          class="text-gray-500 text-sm mt-1"
        >
          user@mail.com
        </p>
        <div
          class="flex items-center justify-center sm:justify-start gap-1 mt-3"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-star text-yellow-400 fill-yellow-400"
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="lucide lucide-star text-yellow-400 fill-yellow-400"
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="lucide lucide-star text-yellow-400 fill-yellow-400"
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="lucide lucide-star text-yellow-400 fill-yellow-400"
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="lucide lucide-star text-yellow-300"
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
            />
          </svg>
          <span
            class="ml-2 text-sm text-gray-600"
          >
            4.5
            /5.0
          </span>
        </div>
        <div
          class="mt-5 text-sm text-gray-700 space-y-2"
        >
          <p>
            <span
              class="font-semibold"
            >
              Role:
            </span>
             Tour Guide
          </p>
          <p>
            <span
              class="font-semibold"
            >
              Status:
            </span>
             Active
          </p>
          <p>
            <span
              class="font-semibold"
            >
              Tour Guide Since:
            </span>
             2024
          </p>
          <p>
            <span
              class="font-semibold"
            >
              Languages:
            </span>
             
            <span
              class="inline-flex flex-wrap gap-2 mt-1"
            >
              <span
                class="bg-blue-100 text-blue-800 text-xs font-medium px-2 py-1 rounded-full"
              >
                English
              </span>
              <span
                class="bg-blue-100 text-blue-800 text-xs font-medium px-2 py-1 rounded-full"
              >
                Bahasa Indonesia
              </span>
              <span
                class="bg-blue-100 text-blue-800 text-xs font-medium px-2 py-1 rounded-full"
              >
                French
              </span>
            </span>
          </p>
        </div>
        <div
          class="mt-6"
        />
      </div>
    </div>
  </div>
</div>
`;
